src/ray/core_worker/core_worker.h

```diff
@@ -256,6 +262,18 @@ class CoreWorkerProcess {
 
   /// To protect accessing the `workers_` map.
   mutable absl::Mutex worker_map_mutex_;
+
+  /// Event loop where the metric exporting are handled. e.g. async gRPC operations.
+  static boost::asio::io_service stats_io_service_;
+
+  /// Stats service worker thread and it will be detached.
```

Comment:
Is there a reason why this should be detached? I assume we're cleaning it up when `enable_stats_count_` reaches `0`.

Reply:
> Is there a reason why this should be detached? I assume we're cleaning it up when `enable_stats_count_` reaches `0`.

Since no shutdown api in opencensus stats, we cannot control all of threads in opencensus and stats io thread might be crashed when unit test process exits.
For sake of using stats safely, we add new shutdown api for opencensus in patches.  After that, no thread will be detached not. I'll delete these comments.